src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -566,7 +565,9 @@ def __init__(self, config):
         self.n_head = config.n_head
 
         # Embedding + LN Embedding
-        self.word_embeddings = nn.Embedding(config.vocab_size, self.embed_dim)
+        self.word_embeddings = nn.Embedding(
+            config.vocab_size, self.embed_dim, dtype=torch.float if config.word_embeddings_in_fp32 else None
```

Comment:
From what I understand the flag `config.word_embeddings_in_fp32` will force the embeddings weight to be in float32 whatever the `torch_dtype` is selected by the user. The default for this option is therefore a breaking change, it should be False to preserve the current behavior.

This is also coded in a very unreadable way (it's only when I reached the tests of this PR that I fully understood what's happening). First the flag should probably be named `force_word_embeddings_in_fp32`. Then the actual implementation should be done in the post init of the model instead of relying on the magical initialization. As it is right now, this won't work with the `device_map` option which will set the weights in the precision passed by `torch_dtype` in `from_pretrained`.